### PR TITLE
small refactoring in payload crate

### PR DESCRIPTION
**Pull request by tcoratger**

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]
lgtm

File: crates/payload/builder/src/database.rs [hunk 1]
```diff
@@ -72,36 +71,27 @@ impl<'a, DB: DatabaseRef> Database for CachedReadsDbMut<'a, DB> {
     type Error = <DB as DatabaseRef>::Error;
 
     fn basic(&mut self, address: Address) -> Result<Option<AccountInfo>, Self::Error> {
-        let basic = match self.cached.accounts.entry(address) {
+        Ok(match self.cached.accounts.entry(address) {
```
Comment (mattsse) [inline crates/payload/builder/src/database.rs:74]: imo these large Ok make this less readable

File: crates/payload/builder/src/service.rs [hunk 2]
```diff
@@ -288,17 +288,14 @@ where
...
-        if attributes.is_none() {
-            trace!(%id, "no matching payload job found to get attributes for");
-        }
-
-        attributes
+            .map(|(j, _)| j.payload_attributes())
+            .or_else(|| {
+                trace!(%id, "no matching payload job found to get attributes for");
+                None
+            })
```
Comment (mattsse) [inline crates/payload/builder/src/service.rs:298]: this is less readable imo

File: crates/payload/builder/src/optimism.rs [hunk 3]
```diff
@@ -26,40 +26,35 @@ impl PayloadBuilderAttributes for OptimismPayloadBuilderAttributes {
...
+                suggested_fee_recipient: attributes.payload_attributes.suggested_fee_recipient,
+                prev_randao: attributes.payload_attributes.prev_randao,
+                withdrawals: attributes
+                    .payload_attributes
+                    .withdrawals
+                    .map(|withdrawals| {
+                        Withdrawals::new(
+                            withdrawals
+                                .into_iter()
+                                .map(convert_standalone_withdraw_to_withdrawal) // Removed the parentheses here
```
Comment (mattsse) [inline crates/payload/builder/src/optimism.rs:51]: this comment can be removed

File: crates/payload/builder/src/database.rs [hunk 4]
```diff
@@ -72,36 +71,27 @@ impl<'a, DB: DatabaseRef> Database for CachedReadsDbMut<'a, DB> {
...
                 entry.insert(CachedAccount::new(self.db.basic_ref(address)?)).info.clone()
             }
-        };
-        Ok(basic)
+        })
     }
 
     fn code_by_hash(&mut self, code_hash: B256) -> Result<Bytecode, Self::Error> {
-        let code = match self.cached.contracts.entry(code_hash) {
+        Ok(match self.cached.contracts.entry(code_hash) {
```
Comment (mattsse) [inline crates/payload/builder/src/database.rs:83]: less readable

**Review by mattsse — CHANGES_REQUESTED** [review]
not a fan of large Ok( blocks, find them less readable

**Review by mattsse — APPROVED** [review]

**Pull request merged**